pkg/scheduler/core/spreadconstraint/select_clusters.go

```diff
@@ -30,14 +30,20 @@ func SelectBestClusters(placement *policyv1alpha1.Placement, groupClustersInfo *
 func selectBestClustersBySpreadConstraints(spreadConstraints []policyv1alpha1.SpreadConstraint,
 	groupClustersInfo *GroupClustersInfo, needReplicas int32) ([]*clusterv1alpha1.Cluster, error) {
 	spreadConstraintMap := make(map[policyv1alpha1.SpreadFieldValue]policyv1alpha1.SpreadConstraint)
+	hasSpreadByLabel := false
 	for i := range spreadConstraints {
 		spreadConstraintMap[spreadConstraints[i].SpreadByField] = spreadConstraints[i]
+		if len(spreadConstraints[i].SpreadByLabel) > 0 {
+			hasSpreadByLabel = true
+		}
 	}
 
 	if _, exist := spreadConstraintMap[policyv1alpha1.SpreadByFieldRegion]; exist {
 		return selectBestClustersByRegion(spreadConstraintMap, groupClustersInfo)
 	} else if _, exist := spreadConstraintMap[policyv1alpha1.SpreadByFieldCluster]; exist {
 		return selectBestClustersByCluster(spreadConstraintMap[policyv1alpha1.SpreadByFieldCluster], groupClustersInfo, needReplicas)
+	} else if hasSpreadByLabel {
+		return selectBestClustersByClusterLabels(spreadConstraints, groupClustersInfo, needReplicas)
```

Comment:
Grouping by label and field cannot take effect at the same time?

Such as the below config:

```yaml
...
    spreadConstraints:
    - maxGroups: 2
      minGroups: 2
      spreadByLabel: stage
    - maxGroups: 2
      minGroups: 2
      spreadByField: cluster
    ...
```

Reply:
https://github.com/karmada-io/karmada/blob/35f3e3f3065235347d2b0420bb29ead08b433b72/pkg/util/validation/validation.go#L100-L103

Reply:
The only thing that's forbidden is the same constraint, they're in two constraints in the example.

Reply:
Thank you for the reminder. Currently, spreadByField is being ignored in this situation. Do you have any good suggestions?

Reply:
I'm sorry, I don't have a good idea yet. I feel that there are many factors to consider when implementing, and it may be complicated to implement.